torch_xla/runtime.py

```diff
@@ -44,10 +45,15 @@ def _maybe_select_default_device():
   if torch_xla._found_libtpu and tpu.num_available_chips() > 0:
     logging.warning('libtpu.so and TPU device found. Setting PJRT_DEVICE=TPU.')
     os.environ[xenv.PJRT_DEVICE] = 'TPU'
-  # TODO(wcromar): Detect GPU device
   elif xu.getenv_as(xenv.GPU_NUM_DEVICES, int, 0) > 0:
     logging.warning('GPU_NUM_DEVICES is set. Setting PJRT_DEVICE=CUDA')
     os.environ[xenv.PJRT_DEVICE] = 'CUDA'
+  elif torch.cuda.is_available():
```

Comment:
What torch.cuda.is_available() does is that it checks both if there is underlying CUDA device and if torch is compiled with USE_CUDA enabled.

Ideally I was thinking we should the same: check the hardware and if torch_xla is compiled with XLA_CUDA enabled. `torch.cuda.is_available()` checks the former for you but I didn't know how to check the latter.

Reply:
btw, do you know how JAX does it?

Reply:
I briefly looked at JAX. From what I can tell, for C++ classes, they have `#ifdef JAX_GPU_CUDA`. For Python, it looks like they try to find the CUDA plugin - https://github.com/google/jax/blob/516b75dc24d8e974d40fbc34e0945ad4aaeae474/jax_plugins/cuda/__init__.py#L30

Do we need to figure all that out for this PR?

Reply:
Great finding! Later, after PyTorch/XLA start using GPU plugin (cc @will-cromar), we can use this way to check if torch_xla is compiled with cuda enabled and add a check at this place. Feel free to do in a follow-up PR.

Reply:
Sounds good, thank you!